pkg/controller/metrics/metrics.go

```diff
@@ -192,9 +219,16 @@ func (mc *Calculator) Start(stopCh <-chan struct{}) error {
 					MetricClusterDeploymentDeprovisioningUnderwaySeconds.WithLabelValues(
 						cd.Name,
 						cd.Namespace,
-						GetClusterDeploymentType(&cd)).Set(
+						clusterType).Set(
 						time.Since(cd.CreationTimestamp.Time).Seconds())
 				}
+
+				if paused, err := strconv.ParseBool(cd.Annotations[constants.SyncsetPauseAnnotation]); err == nil && paused {
+					MetricClusterDeploymentSyncsetPaused.WithLabelValues(
+						cd.Name,
+						cd.Namespace,
+						clusterType).Set(1.0)
+				}
```

Comment:
I might be mistaken, but I think you can just do the Delete in an else right here, does that work? I fear my link misled you to plug it into a controller, whereas in this case I think it's safe for us to clear it right here. (was intended to demonstrate how to Delete for specific labels)